packages/api/src/routes/nfts-check.js

```diff
@@ -5,19 +5,50 @@ import { DBClient } from '../utils/db-client'
 import { parseCid } from '../utils/utils.js'
 import { toCheckNftResponse } from '../utils/db-transforms.js'
 
+const CACHE_MAX_AGE_NO_DEAL_YET = 10 * 60 // in seconds (10 minutes)
 const db = new DBClient(database.url, secrets.database)
 
 /** @type {import('../bindings').Handler} */
 export const nftCheck = async (event, { params }) => {
+  const cache = await caches.open('nft:check')
+  // @ts-ignore match function not found in type https://developer.mozilla.org/en-US/docs/Web/API/Cache
+  let res = await caches.match(event.request.url)
+
+  if (res) {
+    return res
+  }
+
   const cid = parseCid(params.cid)
   const content = await db.getContent(cid.contentCid)
 
-  if (content) {
-    return new JSONResponse({
-      ok: true,
-      value: toCheckNftResponse(cid.sourceCid, content),
-    })
-  } else {
+  if (!content) {
     throw new HTTPError('NFT not found', 404)
   }
+
+  const nftCheckResponseValue = toCheckNftResponse(cid.sourceCid, content)
+  const headers =
+    !!nftCheckResponseValue.deals.find((d) => d.status === 'active') &&
+    nftCheckResponseValue.pin.status === 'pinned'
+      ? {
+          // cache status response with max age
+          'Cache-Control': `public, max-age=${CACHE_MAX_AGE_NO_DEAL_YET}`,
+        }
+      : undefined // cache default
```

Comment:
What is the default though?

Reply:
We should really be specific about the cache period. In this case it's unknown when another deal for the CID may become available so I expect we'd only cache for a relatively short period of time...

Was there an observed perf issue here? This makes the route more complicated and it's used in our monitoring so could mask any underlying issue...

Reply:
Per https://developers.cloudflare.com/workers/runtime-apis/cache/#headers and https://developers.cloudflare.com/cache/how-to/configure-cache-status-code/#edge-ttl default would be 120 minutes.

We can probably use default if we have at least a deal instead?

> Was there an observed perf issue here? This makes the route more complicated and it's used in our monitoring so could mask any underlying issue...

Yesterday we had some timeouts for instance, I don't have empirical answers for this at the moment. The main motivation for this is the work in https://github.com/nftstorage/nft.storage/pull/1386 which will increase significantly the calls to check API

Reply:
If a CID is pinned and in has deals, then some caching of the repsonse would be sensible. Otherwise, caching should be minimal or avoided here.

I'm against landing #1386 as I dont want us to hit the nft.storage api for every single CID fetched from the gateway.